examples/server.config.example

```diff
@@ -39,6 +39,19 @@ instances {
 
     # limit for CAS total content size in bytes
     cas_max_size_bytes: 1073741824 # 1024 * 1024 * 1024
+
+    # an imposed action-key-invariant timeout used in the unspecified timeout case
+    default_action_timeout: {
+      seconds: 600
+      nanos: 0
+    }
+
+    # a limit on the action timeout specified in the action, above which
+    # the operation will report a failed result immediately
+    maximum_action_timeout: {
+      seconds: 3600
```

Comment:
Interesting! We had the opposite problem -- for a while, our system had a minimal supported timeout. Also, note 1h is quite restrictive, integration tests often run longer.

Also, not for this patch, but it seems that our systems are interpreting the timeout field differently. We have not meant for the timeout to  be an absolute expiry deadline -- rather, the clock should only start ticking when actual execution begins: https://github.com/googleapis/googleapis/blob/master/google/devtools/remoteexecution/v1test/remote_execution.proto#L343 It should not count scheduling time, bot creation/setup time, nor any other time required by the service.
If we want to add a limit on this "exec overhead time" (we have considered it), we will put it in a different field (probably in ExecuteRequest), and return RESOURSE_EXHAUSTED if it runs over, so that Bazel can retry.

Reply:
Whelp, that's why it's a config, and just an example - figured it was as good a marker as any, and that if you ran into it, you would be able to figure out what that limit was easily.

I don't think we are interpreting it differently - the fuse is lit when the worker transitions the operation to the Executing state, and when the timeout + config.operation_completed_delay has elapsed, the operationTimeoutDelay watchdog fires its runnable, the operation is forcibly completed with a -1 exit status and a stderr message indicating the failure due to timeout.
Since my pipeline does not change the status to EXECUTING until we have begun the Command, we should have the same measurements, except perhaps for the time spent in ReportResultsStage.

If you do set up a timeout like that, let me know.  Right now there is no limit on a properly functioning worker's time to setup for execution, as long as they are polling regularly enough. The only reaction to failure to poll is requeuing, not operation failure.

Reply:
Oh, sorry, you're right, I didn't get when the fuse was lit. There's one thing that bugs me about expireOperation function -- I don't see you doing executeResponse.setStatus(Status.newBuilder().setCode(DEADLINE_EXCEEDED)). That's the way Bazel will know about the timeout, and display the test status as TIMEOUT rather than FAILED. You could also use the operation.error field, but that is deprecated, so you shouldn't (because it's a oneof, so if you use error, you cannot return the result).

Reply:
Gotcha. I threw that in there, but I assume there are numerous other places where I'm missing this style of response - I had filed #73 to deal with this independently